src/reference-manual/compiler-optimizations.Rmd

```diff
@@ -0,0 +1,487 @@
+The Stanc3 compiler can attempt to optimize a Stan program as it is compiled. The optimized program has the same behavior as the unoptimized program, but it may be faster, more memory efficient, or more numerically stable.
+
+This section introduces the available optimization options and describes their effect.
+
+You can see a printout of a representation of the Stan program after the optimizations have been applied with the Stanc3 command-line option `--debug-optimized-mir-pretty`. You can see an analogous representation of the program before optimizations have been applied with `--debug-transformed-mir-pretty`.
```

Comment:
```suggestion
You can see a pretty-printed representation of the Stan program after optimizations with the Stanc3 command-line option `--debug-optimized-mir-pretty`. To print an analogous representation of the program before optimizations use the `--debug-transformed-mir-pretty` flag.
```

Reply:
I would maybe move this below to not confuse the reader. I doubt many will try this, but I bet many will be interested in the details of optimizations.